src/python/pants/backend/python/dependency_inference/import_parser.py

```diff
@@ -33,56 +34,41 @@ class ParsedPythonImports:
     def all_imports(self) -> FrozenOrderedSet[str]:
         return FrozenOrderedSet(sorted([*self.explicit_imports, *self.inferred_imports]))
 
-    @classmethod
-    def empty(cls) -> "ParsedPythonImports":
-        return cls(FrozenOrderedSet(), FrozenOrderedSet())
-
-
-class VisitorInterface(Protocol):
-    explicit_imports: Set[str]
-    inferred_imports: Set[str]
 
-
-def parse_file(*, filename: str, content: str, module_name: str) -> Optional[VisitorInterface]:
-    """Parse the file for python imports, and return a visitor with the imports it found."""
-    # Parse all python 3 code with libCST. We parse assuming python 3 goes first, because we assume
-    # most user code will be python 3.
-    # TODO(#10921): identify the appropriate interpreter version beforehand!
+def parse_file(*, filename: str, content: str) -> Optional[Tuple]:
     try:
-        # NB: Since all python 3 code is forwards-compatible with the 3.8 parser, and the import
-        # syntax remains unchanged, we are safely able to use the 3.8 parser for parsing imports.
-        # TODO(#10922): Support parsing python 3.9/3.10 with libCST!
-        config = cst.PartialParserConfig(python_version="3.8")
-        cst_tree = cst.parse_module(content, config=config)
-        completed_cst_visitor = _CSTVisitor.visit_tree(cst_tree, module_name=module_name)
-        return completed_cst_visitor
-    except cst.ParserSyntaxError as e:
-        # NB: When the python 3 ast visitor fails to parse python 2 syntax, it raises a
-        # ParserSyntaxError. This may also occur when the file contains invalid python code. If we
-        # successfully parse a python 2 file with a python 3 parser, that should not change the
-        # imports we calculate.
-        logger.debug(f"Failed to parse {filename} with python 3.8 libCST parser: {e}")
-
-    try:
-        py27_tree = ast27.parse(content, filename=filename)
-        completed_ast27_visitor = _Py27AstVisitor.visit_tree(py27_tree, module_name=module_name)
-        return completed_ast27_visitor
-    except SyntaxError as e:
-        logger.debug(f"Failed to parse {filename} with python 2.7 typed-ast parser: {e}")
-
-    return None
+        # NB: The Python 3 ast is generally backwards-compatible with earlier versions. The only
+        # breaking change is `async` `await` becoming reserved keywords in Python 3.7 (deprecated
+        # in 3.6). If the std-lib fails to parse, we could use typed-ast to try parsing with a
+        # target version of Python 3.5, but we don't because Python 3.5 is almost EOL and has very
+        # low usage.
+        # We will also fail to parse Python 3.8 syntax if Pants is run with Python 3.6 or 3.7.
+        # There is no known workaround for this, beyond users changing their `./pants` script to
+        # always use >= 3.8.
```

Comment:
Is there a way to detect this condition and flag it to a user? 

Also I assume this is documented?

Reply:
Would a package like https://pypi.org/project/ast-compat/ be useful?

Reply:
> Would a package like https://pypi.org/project/ast-compat/ be useful?

Something like that would be good, but that doesn't look production ready nor can I figure out how to use it. Generally, most projects use typed-ast, which works great expect that they decided to not add Py38 support to it. I've heard of libCST and RedBaron too, which are both too slow. 

Unfortunately, I think the best we can do is typed-ast, or running with Py38. (If run with Py38, it supports parsing all prior Py3 versions)